### PR TITLE
Added ChatGptService

### PR DESCRIPTION
**Pull request by Suleman70**
**Integrating ChatGPT to TJ-Plays**

I added the ChatGptService and  ~~AIResponseParser~~ (is now Removed) class from the TJ-Bot repository to TJ-Plays and also added its dependencies.

**TODO:** We need to come up with a way of making sure there is only one instance of this class - and that it can be shared among other classes.

### CODE REVIEW

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java:1]: this is not the latest file use the current [ChatGptService](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/features/chatgpt/ChatGptService.java)

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/AIResponseParser.java
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/AIResponseParser.java:1]: This is not in TJ-Bot codebase

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/AIResponseParser.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a class to partition long text blocks into smaller blocks which work with Discord's
+ * API. Initially constructed to partition text from AI text generation APIs.
+ * This class is from the TJ-Bot repository: <a href="https://github.com/Together-Java/TJ-Bot">...</a>
```
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/AIResponseParser.java:14]: dont

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Service used to communicate to OpenAI API to generate responses.
+ * This class is from the TJ-Bot repository: <a href="https://github.com/Together-Java/TJ-Bot">...</a>
```
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java:21]: dont

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Comment by Suleman70** [comment]
Added the changes @Taz03

File: app/src/main/java/com/togetherjava/tjplays/Bot.java
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/Bot.java:1]: revert the changes of this file

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        if (apiKey.isBlank() || keyIsDefaultDescription) {
+            isDisabled = true;
+            return;
+        }
+
+        openAiService = new OpenAiService(apiKey, TIMEOUT);
+
+        ChatMessage setupMessage = new ChatMessage(ChatMessageRole.SYSTEM.value(), """
+                For code supplied for review, refer to the old code supplied rather than
+                rewriting the code. DON'T supply a corrected version of the code.\s""");
```
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java:69]: we dont need this, in this project so remove it

File: app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+     */
+    public Optional<String> ask(String question, String context) {
+        if (isDisabled) {
+            return Optional.empty();
+        }
+
+        try {
+            String instructions = "KEEP IT CONCISE, NOT MORE THAN 280 WORDS";
+            String questionWithContext = "context: Category %s on a Java Q&A discord server. %s %s"
+                    .formatted(context, instructions, question);
```
Comment (Taz03) [inline app/src/main/java/com/togetherjava/tjplays/services/chatgpt/ChatGptService.java:101]: we dont need these instructions

File: app/build.gradle [hunk 5]
```diff
@@ -40,7 +44,10 @@ shadowJar {
 
 dependencies {
     implementation 'net.dv8tion:JDA:5.0.0-beta.21'
-
+    implementation "com.theokanning.openai-gpt3-java:api:$chatGPTVersion"
+    implementation "com.theokanning.openai-gpt3-java:service:$chatGPTVersion"
+    implementation 'org.apache.logging.log4j:log4j-core:2.23.0'
+    runtimeOnly 'org.apache.logging.log4j:log4j-slf4j18-impl:2.18.0'
```
Comment (Taz03) [inline app/build.gradle:50]: add new lines in between

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Review by Taz03 — DISMISSED** [review]

**Review by Taz03 — APPROVED** [review]

**Pull request merged**